### ISSUE TITLE
No internet on certain WiFi connection

### ISSUE THREAD

**Issue by codedmon** [trailofbits/algo#1700]
Setup:
- Running Algo on an DigitalOcean droplet.  Connect to it via WireGuard Android app.

Issue:
- I am able to connect to my Algo VPN via my mobile connection (Verizon) and my home WiFi, no issues.   My work has (2) wireless access points (WAP).  The first WAP is for company owned devices and the second is a "guest" WAP that visitors can use.   Employees also use the second WAP with their personal devices.   When I connect to the WAP, I have no internet connection.  The WireGuard Android app is very plain, not many features and options.  I don't even see a real-time log so I can see what might be going on when connected to this particular WAP.

I suspect that this WAP is blocking the connection to my VPN (port?).  Any suggestions on troubleshooting this?

**Comment by TC1977** [comment]
I don't have any experience with the Android WireGuard app. The iOS app has an option in the settings screen to "View log".

What if you try using an IPsec client?

**Comment by codedmon** [comment]
> I don't have any experience with the Android WireGuard app. The iOS app has an option in the settings screen to "View log".
> 
> What if you try using an IPsec client?

I read on another thread that IPSEC support was depreciated for Android.

**Comment by jackivanov** [comment]
Try to switch WireGuard to 443 or 80 port

**Comment by codedmon** [comment]
> Try to switch WireGuard to 443 or 80 port

How would i go about doing this?   Does this require a new server to be setup or can i edit existing files via SSH?

**Comment by TC1977** [comment]
Check #1707 for an up-to-date discussion of how to do this with an existing server.

But since (I've heard) it's not uncommon for hostile networks to block UDP/51820 or even all UDP, you might be better off trying IPsec (which uses UDP/500 and 4500) or Open VPN (which uses TCP/1194, and can be changed). WireGuard only uses UDP ports, so switching to UDP/80 or UDP/443 might not help much.

**Comment by codedmon** [comment]
> Check #1707 for an up-to-date discussion of how to do this with an existing server.
> 
> WireGuard only uses UDP ports, so switching to UDP/80 or UDP/443 might not help much.

I could run a netstat command to see which ports are open.

**Comment by TC1977** [comment]
Running netstat on your Android will only tell you that WireGuard isn't passing a connection out. Netstat would help a lot more if you can run it on the WAP, to determine that it's actually passing a WireGuard connection through. But if you can run netstat on the WAP...

**Comment by codedmon** [comment]
> Running netstat on your Android will only tell you that WireGuard isn't passing a connection out. Netstat would help a lot more if you can run it on the WAP, to determine that it's actually passing a WireGuard connection through. But if you can run netstat on the WAP...

I'm a newb when I'm comes to networking.  I googled how to see which ports are open and assumed i could run the netstat command on a Windows machine hooked to the WAP.  Is there another way to find if the WAP allows UDP connections?

**Comment by TC1977** [comment]
Well, WhatsApp and FaceTime (to give two examples) use UDP for streaming. Are those blocked?

**Comment by Thomvh** [comment]
I have the same problem, ipsec ike isn't blocked on the network but Wireguard and Openvpn etc are all blocked. This means a macbook can get on the vpn. But my windows machine I can't get to work. Strongswan supports windows, but algo doesn't.

**Comment by codedmon** [comment]
> Well, WhatsApp and FaceTime (to give two examples) use UDP for streaming. Are those blocked?

I don't use either of those apps.   I did run the netstat command on the network and none of the UDP ports mentioned were shown.

**Comment by dguido** [comment]
Hi @codedmon,

Thank you for reporting this WireGuard issue. Since you reported this, WireGuard has reached stable 1.0.0 release (Spring 2020) and has undergone substantial security review.

WireGuard is now:
- Officially stable and production-ready
- Included in the Linux kernel
- Actively maintained with regular updates
- The recommended VPN protocol for Algo

The connection issues you experienced were likely related to the pre-1.0 development versions. These have been resolved in the stable releases.

Since WireGuard has matured significantly and this issue relates to early development versions, I'm closing it. If you encounter any issues with current versions of WireGuard and Algo, please feel free to open a new issue.

Thank you for your patience and for using Algo!
